### PR TITLE
Add navigation section and document 3D navigation (3.4)

### PR DESCRIPTION
**Pull request by skyace65**
Adds a navigation section to the documentation as suggested in #4135 and creates a page for 3D navigation. I'm pushing to 3.4 because this doesn't work on the master branch since the nodes were changed, and this links to the class reference. I'll create a modified version for 4.0 sometime this week. All images have gone through png compression.

Also this tutorial is heavily dependent on the navigation mesh class reference page being updated for 3.4 now that it's fully documented.

### CODE REVIEW

**Comment by skyace65** [comment]
Does anyone know what's going on with the CI error? All of the links look like they're formatted properly to me but something has to be wrong.

**Comment by skyace65** [comment]
Fixed it.

File: tutorials/navigation/real_time_navigation_3d.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Real Time Navigation (3D)
+=========================
+
+Introduction
+------------
+
+Godot provides two nodes which can be used to achieve real time navigation
+in a 3D environment, :ref:`Navigation <class_Navigation>` and :ref:`NavigationMeshInstance <class_NavigationMeshInstance>`.
+
+A navigation mesh instance is used to hold a :ref:`Navigation <class_NavigationMesh>`,
```
Comment (mhilbrunner) [inline tutorials/navigation/real_time_navigation_3d.rst:12]:
```suggestion
A navigation mesh instance is used to hold a :ref:`NavigationMesh <class_NavigationMesh>`,
```

File: tutorials/navigation/real_time_navigation_3d.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+which defines what ares in a scene can be navigated to. The navigation
```
Comment (mhilbrunner) [inline tutorials/navigation/real_time_navigation_3d.rst:13]:
```suggestion
which defines what ares in a scene can be navigated to. The :ref:`Navigation <class_Navigation>`
```
(linking it once more for clarity, and to mark the difference to NavigationMesh in the sentence before)

**Review by mhilbrunner — COMMENTED** [review]
Looks good, some small notes.

Also, we should add the word "pathfinding" somewhere on this doc for google-ability IMO :)

**Comment by skyace65** [comment]
Fixed. I updated the navigation node description to use the word pathfinding

**Comment by mhilbrunner** [comment]
Thanks for this :)

**Pull request merged**